src/pages/api/callAssignments/targets.ts

```diff
@@ -36,6 +36,35 @@ export default async function handler(
   );
   const targetsData = await targetsRes.json();
 
+  const blockedTargets = targetsData.data.filter(
+    (target: ZetkinTarget) => target.status.block_reasons.length > 0
+  );
+  const organizerActionNeeded: number = blockedTargets.filter(
+    (target: ZetkinTarget) =>
+      target.status.block_reasons.includes('organizer_action_needed')
+  ).length;
+
+  const missingPhoneNumber: number = blockedTargets.filter(
+    (target: ZetkinTarget) =>
+      target.status.block_reasons.includes('no_number') &&
+      !target.status.block_reasons.includes('organizer_action_needed')
+  ).length;
+
+  const callBackLater: number = blockedTargets.filter(
+    (target: ZetkinTarget) =>
+      target.status.block_reasons.includes('call_back_after') &&
+      !target.status.block_reasons.includes('organizer_action_needed') &&
+      !target.status.block_reasons.includes('no_number')
+  ).length;
+
+  const calledTooRecently: number = blockedTargets.filter(
+    (target: ZetkinTarget) =>
+      target.status.block_reasons.includes('cooldown') &&
+      !target.status.block_reasons.includes('call_back_after') &&
+      !target.status.block_reasons.includes('organizer_action_needed') &&
+      !target.status.block_reasons.includes('no_number')
+  ).length;
+
```

Comment:
I feel like this code is written in a way that has several drawbacks, compared to just doing a simple loop with conditionals inside. Something like this:

```ts
let organizerActionNeeded = 0,
    missingPhoneNumber = 0,
    callBackLater = 0,
    calledTooRecently = 0;

blockedTargets.forEach(target => {
    const reasons = target.status.block_reasons;
    if (reasons.includes('organizer_action_needed')) {
        organizerActionNeeded++;
    } else if (reasons.includes('no_number')) {
        missingPhoneNumber++;
    } // ...
});
```

Doing it that way should be faster, more readable, and less error prone. Faster because there are less loops, more readable because the relevant condition stands out and the fact that the conditions are related is more obvious, and less error prone because the if/else approach makes it impossible to count the same target twice if there's a mistake in the conditions.